libs/sdk-core/src/breez_services.rs

```diff
@@ -1012,6 +1022,19 @@ impl BreezServices {
                 Ok(payment)
             }
             Err(e) => {
+                if let Some(invoice) = invoice.clone() {
+                    // Only persist errors specific to a node or route related failure
+                    match e.clone() {
+                        SendPaymentError::PaymentFailed { err }
```

Comment:
Perhaps we should persist any error? I am concerned we will miss errors when we add more errors to the enum.
Also worst case we will add an entry here without a failure in greenlight that will never be used.
What do you think?

Reply:
Agreed and done. We can always refine later.